test/expected/insert_single.out

```diff
@@ -339,3 +344,65 @@ SELECT * FROM "3dim";
  Fri Jan 20 09:00:21 2017 | 21.2 | brown  | sthlm
 (3 rows)
 
+-- test that explain works
+EXPLAIN
+INSERT INTO "3dim" VALUES('2017-01-21T09:00:01', 32.9, 'green', 'nyc'),
+                         ('2017-01-21T09:00:47', 27.3, 'purple', 'la') RETURNING *;
+                                   QUERY PLAN                                   
+--------------------------------------------------------------------------------
+ Custom Scan (HypertableInsert)  (cost=0.00..0.03 rows=2 width=80)
+   ->  Insert on "3dim"  (cost=0.00..0.03 rows=2 width=80)
+         ->  Custom Scan (ChunkDispatch)  (cost=0.00..0.03 rows=2 width=80)
+               ->  Values Scan on "*VALUES*"  (cost=0.00..0.03 rows=2 width=80)
+(4 rows)
+
+EXPLAIN
+WITH "3dim_insert" AS (
+     INSERT INTO "3dim" VALUES('2017-01-21T09:01:44', 19.3, 'black', 'la') RETURNING time, temp
+), regular_insert AS (
+   INSERT INTO regular_table VALUES('2017-01-21T10:00:51', 14.3) RETURNING time, temp
+) INSERT INTO "1dim" (SELECT time, temp FROM "3dim_insert" UNION SELECT time, temp FROM regular_insert);
+                                            QUERY PLAN                                             
+---------------------------------------------------------------------------------------------------
+ Custom Scan (HypertableInsert)  (cost=0.09..0.12 rows=2 width=16)
+   ->  Insert on "1dim"  (cost=0.09..0.12 rows=2 width=16)
+         CTE 3dim_insert
+           ->  Custom Scan (HypertableInsert)  (cost=0.00..0.01 rows=1 width=80)
+                 ->  Insert on "3dim"  (cost=0.00..0.01 rows=1 width=80)
+                       ->  Custom Scan (ChunkDispatch)  (cost=0.00..0.01 rows=1 width=80)
+                             ->  Result  (cost=0.00..0.01 rows=1 width=80)
+         CTE regular_insert
+           ->  Insert on regular_table  (cost=0.00..0.01 rows=1 width=16)
+                 ->  Result  (cost=0.00..0.01 rows=1 width=16)
+         ->  Custom Scan (ChunkDispatch)  (cost=0.07..0.09 rows=2 width=16)
+               ->  Unique  (cost=0.07..0.09 rows=2 width=16)
+                     ->  Sort  (cost=0.07..0.08 rows=2 width=16)
+                           Sort Key: "3dim_insert"."time", "3dim_insert".temp
+                           ->  Append  (cost=0.00..0.06 rows=2 width=16)
+                                 ->  CTE Scan on "3dim_insert"  (cost=0.00..0.02 rows=1 width=16)
+                                 ->  CTE Scan on regular_insert  (cost=0.00..0.02 rows=1 width=16)
+(17 rows)
+
+-- test prepared statement INSERT
+PREPARE "1dim_plan" (timestamp, float) AS
+    INSERT INTO "1dim" VALUES($1, $2) ON CONFLICT (time) DO NOTHING;
+EXECUTE "1dim_plan" ('2017-04-17 23:35', 31.4);
```

Comment:
So I know that in the prepared path with bound params it takes at least 5 calls (executes) for it to create the generic plan for selects. Not sure if its the same for Inserts. But somebody should probably trace this through in lldb. The function to look at is `choose_custom_plan`. We want to check both the prepared custom and generic plans I guess. There is also a  CURSOR_OPT_GENERIC_PLAN which should force that and I guess is a cursor option but no idea how to force that in regression tests. Maybe look at postgres regression tests?